thirdparty/onnx/CMakeLists.txt

```
@@ -44,20 +44,19 @@ foreach(_onnx_target onnx onnx_proto)
         $<TARGET_PROPERTY:protobuf::libprotobuf,INTERFACE_INCLUDE_DIRECTORIES>)
 endforeach()
 
-if(WIN32)
-    # from onnx==1.13.1 it requires C++17 when compiling on Windows
-    target_compile_features(onnx PRIVATE cxx_std_17)
+if(WIN32 AND MINGW)
```

```suggestion
if(MINGW)
```
because `MINGW` implies Windows